spot_driver/src/api/default_spot_api.cpp

```diff
@@ -14,28 +14,44 @@
 
 namespace spot_ros2 {
 
-DefaultSpotApi::DefaultSpotApi(const std::string& sdk_client_name)
-    : client_sdk_{::bosdyn::client::CreateStandardSDK(sdk_client_name)} {}
+DefaultSpotApi::DefaultSpotApi(const std::string& sdk_client_name, const std::optional<std::string>& certificate) {
+  if (certificate.has_value()) {
+    client_sdk_ = std::make_unique<::bosdyn::client::ClientSdk>();
+    client_sdk_->SetClientName(sdk_client_name);
+    if (const auto status = client_sdk_->LoadRobotCertFromFile(certificate.value()); !status) {
+      throw std::runtime_error(status.message());
+    }
+    client_sdk_->Init();
+  } else {
+    client_sdk_ = ::bosdyn::client::CreateStandardSDK(sdk_client_name);
+  }
+}
 
-tl::expected<void, std::string> DefaultSpotApi::createRobot(const std::string& ip_address,
-                                                            const std::string& robot_name) {
+tl::expected<void, std::string> DefaultSpotApi::createRobot(const std::string& robot_name,
+                                                            const std::string& ip_address,
+                                                            const std::optional<int>& port) {
   robot_name_ = robot_name;
 
-  auto create_robot_result = client_sdk_->CreateRobot(ip_address);
+  auto create_robot_result = client_sdk_->CreateRobot(ip_address, ::bosdyn::client::USE_PROXY);
```

Comment:
This is key to ensure the SDK does not attempt communication over insecure channels when `ip_address` is `localhost`.